pages/api/v1/auth/actions/createOpengovTreasuryProposal.ts

```diff
@@ -19,11 +19,13 @@ const handler: NextApiHandler<CreatePostResponseType> = async (req, res) => {
 	const network = String(req.headers['x-network']);
 	if(!network || !isValidNetwork(network)) return res.status(400).json({ message: 'Invalid network in request header' });
 
-	const { content, title, postId, userId, proposerAddress, tags } = req.body;
+	const { content, title, postId, userId, proposerAddress, tags, postLink } = req.body;
 	if(!content || !title || !proposerAddress) return res.status(400).json({ message: 'Missing parameters in request body' });
 
 	if(isNaN(Number(userId)) || isNaN(Number(postId)))  return res.status(400).json({ message: 'Invalid parameters in request body' });
 
+	if(postLink && isNaN(postLink?.id)) return res.status(400).json({ message: 'Invalid discussion post id' });
```

Comment:
incorrect validation and storage,
post_link is an object : `{id: number, type: string(proposal type)}`